src/MongoDB.Driver/Encryption/ClientEncryption.cs

```diff
@@ -0,0 +1,143 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Libmongocrypt;
+
+namespace MongoDB.Driver
+{
+    /// <summary>
+    /// Explicit client encryption.
+    /// </summary>
+    public class ClientEncryption
+    {
+        // private fields
+        private readonly LibMongoCryptController _libMongoCryptController;
+
+        // constructors
+        internal ClientEncryption(LibMongoCryptController libMongoCryptController)
+        {
+            _libMongoCryptController = Ensure.IsNotNull(libMongoCryptController, nameof(libMongoCryptController));
+        }
+
+        // public methods
+        /// <summary>
+        /// Creates a data key.
+        /// </summary>
+        /// <param name="kmsProvider">The kms provider.</param>
+        /// <param name="dataKeyOptions">The data key options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A data key.</returns>
+        public BsonValue CreateDataKey(string kmsProvider, DataKeyOptions dataKeyOptions, CancellationToken cancellationToken)
+        {
+            return _libMongoCryptController.CreateDataKey(
+                kmsProvider, 
+                dataKeyOptions.AlternateKeyNames, 
+                dataKeyOptions.MasterKey, 
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates a data key.
+        /// </summary>
+        /// <param name="kmsProvider">The kms provider.</param>
+        /// <param name="dataKeyOptions">The data key options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A data key.</returns>
+        public async Task<BsonValue> CreateDataKeyAsync(string kmsProvider, DataKeyOptions dataKeyOptions, CancellationToken cancellationToken)
+        {
+            return await _libMongoCryptController
+                .CreateDataKeyAsync(
+                    kmsProvider,
+                    dataKeyOptions.AlternateKeyNames,
+                    dataKeyOptions.MasterKey, 
+                    cancellationToken)
+                .ConfigureAwait(false);
```

Comment:
Do you need to use await when you aren't going to do anything other than return the result?

You can get rid of the async/await and ConfigureAwait.

Unless you know of some reason to the contrary?